### PR TITLE
[ONNX] Fixed Select symbolic to export slice when index = negative one

### PR DESCRIPTION
**Pull request by neginraoof**
Exporting torch.select when index = negative one (x[:,-1]) was broken. This PR has the fix in symbolic function for select.

### CODE REVIEW

**Review by BowenBao — APPROVED** [review]
LGTM. some minor comments

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -398,10 +398,12 @@ def select(g, self, dim, index):
     if dim > 1:
         # TODO: this is a temporary hack because of the implementation details
```
[the remark this answers is not in the document]
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Thanks, I was a bit confused about whether we want 'axis' as a tensor input. I'll fixed that.

**Comment by dzhulgakov** [comment]
Looks like CI is failing in some related onnx tests

**Comment by neginraoof** [comment]
@dzhulgakov The failure is fixed.
cc @houseroad

File: caffe2/onnx/backend.cc [hunk 2]
```diff
@@ -709,20 +709,16 @@ Caffe2Ops Caffe2Backend::CreateGather(
...
-    BuildOperator(c2_op, "Gather", inputs, outputs);
-  } else if (axis == 1) {
-    BuildOperator(c2_op, "BatchGather", inputs, outputs);
-  } else {
-    CAFFE_THROW(
-        "Caffe2 only supports Gather with axis being 0 or 1, ",
-        "whereas axis is ",
-        axis);
-  }
+  BuildOperator(c2_op, "Gather", inputs, outputs, {arg_axis});
```
Comment (houseroad) [inline caffe2/onnx/backend.cc]: Wait, we didn't extend Gather op correctly... Why you can relax the export?

**Review by houseroad — CHANGES_REQUESTED** [review]
The caffe2 onnx backend changes seem wrong.

File: caffe2/onnx/backend.cc (hunk 2 above)
Comment (neginraoof) [reply caffe2/onnx/backend.cc]: Gather's implementation had been updated in Caffe2 https://github.com/pytorch/pytorch/commit/773f4d8081634884ec421236165bd2ae19417503
I just updated this use case.

**Comment by neginraoof** [comment]
cc @houseroad for review

**Comment by neginraoof** [comment]
cc @houseroad for review

**Comment by neginraoof** [comment]
cc @houseroad for review

**Review by spandantiwari — APPROVED** [review]
LGTM.

**Comment by neginraoof** [comment]
cc @houseroad for review

**Review by houseroad — APPROVED** [review]
Looks good, thanks.

**Comment by neginraoof** [comment]
@houseroad Is this branch merged? I'm not seeing this changes in master

**Comment by houseroad** [comment]
Could you create a new PR with a new branch instead of using this one?

**Comment by neginraoof** [comment]
@houseroad  This change is required as part of this PR: https://github.com/pytorch/pytorch/pull/26549
I think it's better to merge the two. It'll be a bigger PR to review, but since they are dependent I think it'll make sense.

**Pull request closed without merging**